File: desktop/macos/Sources/Localyze/Inference/ModelPath.swift
import Foundation

/// Where the on-disk model bundle lives. Single source of truth for both the
/// downloader (writes here) and the backend (reads here).
enum ModelPath {
    static let supportRoot: URL = {
        let appSupport = FileManager.default.urls(for: .applicationSupportDirectory,
                                                  in: .userDomainMask).first!
        return appSupport.appendingPathComponent("Localyze", isDirectory: true)
    }()

    static let modelDir: URL = supportRoot.appendingPathComponent("models/gemma-4-e4b-it-mlx",
                                                                  isDirectory: true)

    static let artifactsDir: URL = {
        let home = FileManager.default.homeDirectoryForCurrentUser
        return home.appendingPathComponent("Localyze/artifacts", isDirectory: true)
    }()

    static func ensureDirectories() throws {
        for dir in [supportRoot, modelDir, artifactsDir] {
            try FileManager.default.createDirectory(at: dir, withIntermediateDirectories: true)
        }
    }
}

File: desktop/windows/Localyze/Inference/ModelPath.cs
using System.IO;

namespace Localyze.Inference;

/// Single source of truth for on-disk paths. Used by Download/ + Inference/.
public static class ModelPath
{
    public static string AppDataRoot { get; } = Path.Combine(
        System.Environment.GetFolderPath(System.Environment.SpecialFolder.LocalApplicationData),
        "Localyze");

    public static string ModelDir { get; } = Path.Combine(AppDataRoot, "models", "gemma-4-e4b-it-onnx");

    public static string ArtifactsDir { get; } = Path.Combine(
        System.Environment.GetFolderPath(System.Environment.SpecialFolder.UserProfile),
        "Localyze", "artifacts");

    public static void EnsureDirectories()
    {
        Directory.CreateDirectory(AppDataRoot);
        Directory.CreateDirectory(ModelDir);
        Directory.CreateDirectory(ArtifactsDir);
    }

    public static bool ModelExists =>
        File.Exists(Path.Combine(ModelDir, "genai_config.json"));
}
